community/detectors/spring_framework_cve_2022_22965/src/main/java/com/google/tsunami/plugins/detectors/spring/SpringCve202222965Detector.java

```diff
@@ -0,0 +1,183 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.tsunami.plugins.detectors.spring;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.tsunami.common.net.http.HttpRequest.get;
+import static com.google.tsunami.common.net.http.HttpRequest.post;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.util.Timestamps;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.common.net.http.HttpStatus;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Date;
+import javax.inject.Inject;
+
+/**
+ * A {@link VulnDetector} that detects Spring Framework RCE(CVE-2022-22965)
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "SpringCve202222965Detector",
+    version = "0.1",
+    description = "This detector checks for Spring Framework RCE(CVE-2022-22965).",
+    author = "C4o (syttcasd@gmail.com)",
+    bootstrapModule = SpringCve202222965DetectorBootstrapModule.class)
+public final class SpringCve202222965Detector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final String FILENAME = "SpringCoreRCEDetect";
+  private static final String FORMAT = ".yyyy";
+  private static final SimpleDateFormat time_format = new SimpleDateFormat(FORMAT);
+  private static final Date time_now = new Date();
+  private static final String VERIFY_STRING = "TSUNAMI_SpringCoreRCEDetect";
+  private static final String VULNERABILITY_PAYLOAD_STRING = "class.module.classLoader.resources."
```

Comment:
Are you sure? You're going to attack the site directly like that, causing the logging function to fail and there's a backdoor left over? Are you sure you want to test like that?

Reply:
Thanks for your useful advice! I've repalced my payload with a more harmless one  and changed code to use crawler to get more potential point. Could you review my code again?

BTW, have you recently joined Google? Cause it was magl0 reviewing code before.